trucks.js

```diff
@@ -503,9 +505,55 @@ var foodTrucks = [
 	}
 ];
 
-// this module should support the following methods:
 // getTrucks() - return all trucks
+function getTrucks() {
+	return foodTrucks;
+}
+
 // getTruck(name) - return the truck object matching 'name'
+function getTruck(name) {
+	return _.filter(foodTrucks, function(truck) {
+		return _.contains(truck, name);
+	});
+}
+
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
-// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+function getFoodTypes() {
```

Comment:
Right now this is returning all food truck objects, because the function passed to _.filter is always returning a "truthy" value which, in this case, is an array. The getFoodTypes method takes a couple of steps: 1. get a list of all the possible food types - this can be done by looping over all food trucks, then looping over each type array for each food truck object in the foodTrucks list, adding each type to a master list. Another way to do this would be: `function getFoodTypes() { var list = _.map(foodTrucks, function (truck) { return truck.type; }); list = _.flatten(list); return _.uniq(list); }`

The flatten step is necessary because the first _.map step creates and array of arrays and _.flatten will just make it a simple array. Then, _.uniq is called to return an array with any duplicates removed.